jax/version.py

```diff
@@ -32,28 +32,38 @@ def _get_version_string() -> str:
   # In this case we return it directly.
   if _release_version is not None:
     return _release_version
-  try:
-    return _version_from_git_tree(_version)
-  except:
-    return _version_from_todays_date(_version)
+  return _version_from_git_tree(_version) or _version_from_todays_date(_version)
 
 
 def _version_from_todays_date(base_version: str) -> str:
   datestring = datetime.date.today().strftime("%Y%m%d")
   return f"{base_version}.dev{datestring}"
 
 
-def _version_from_git_tree(base_version: str) -> str:
-  stdout = subprocess.check_output(["git", "show", "-s", "--format=%at", "HEAD"])
-  timestamp = int(stdout.decode().strip())
-  datestring = datetime.date.fromtimestamp(timestamp).strftime("%Y%m%d")
-  assert datestring.isnumeric()
-
-  stdout = subprocess.check_output(["git", "describe", "--long", "--always"])
-  commit_hash = stdout.decode().strip().rsplit('-', 1)[-1]
-  assert commit_hash.isalnum()
-
-  return f"{base_version}.dev{datestring}+{commit_hash}"
+def _version_from_git_tree(base_version: str) -> str | None:
+  try:
+    root_directory = os.path.dirname(os.path.realpath(__file__))
+
+    # Get date string from date of most recent git commit.
+    p = subprocess.Popen(["git", "show", "-s", "--format=%at", "HEAD"],
+                         cwd=root_directory,
+                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, _ = p.communicate()
+    timestamp = int(stdout.decode().strip())
+    datestring = datetime.date.fromtimestamp(timestamp).strftime("%Y%m%d")
+    assert datestring.isnumeric()
+
+    # Get commit hash from most recent git commit.
+    p = subprocess.Popen(["git", "describe", "--long", "--always"],
+                         cwd=root_directory,
+                         stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, _ = p.communicate()
+    commit_hash = stdout.decode().strip().rsplit('-', 1)[-1]
+    assert commit_hash.isalnum()
+  except:
```

Comment:
I might be tempted to use a more targeted exception here.

Reply:
I thought of that... the problem is that this runs any time you import jax in the source tree, so if we target the exception but forget about a possibility (e.g. a weird exception class when using `subprocess` on windows), then `import jax` will fail.

What do you think?

Reply:
True, but that's no different than any other exception we might try to catch. No strong feelings.